terraform/source.go

```diff
@@ -195,7 +199,57 @@ func ToSourceUrl(source string, workingDir string) (*url.URL, error) {
 		return nil, errors.WithStackTrace(err)
 	}
 
-	return parseSourceUrl(rawSourceUrlWithGetter)
+	sourceUrl, err := parseSourceUrl(rawSourceUrlWithGetter)
+	if err != nil {
+		return nil, err
+	}
+
+	return PrependSourceType(sourceUrl), nil
+}
+
+// PrependSourceType prepends one of the "s3, gcs3, hg, git" source type to the http(s)/ssh scheme, if the user has omittied it.
```

Comment:
NIT: link to module source URL docs or maybe go-getter docs as context on what these are.

Reply:
Updated the comment. Thanks.